pkg/cloud/libvirt/client/domain.go

```diff
@@ -105,6 +69,53 @@ func newDomainDef() libvirtxml.Domain {
 	return domainDef
 }
 
+func newDevicesDef() *libvirtxml.DomainDeviceList {
+	var serialPort uint
+
+	domainList := libvirtxml.DomainDeviceList{
+		Channels: []libvirtxml.DomainChannel{
+			{
+				Target: &libvirtxml.DomainChannelTarget{
+					VirtIO: &libvirtxml.DomainChannelTargetVirtIO{
+						Name: "org.qemu.guest_agent.0",
+					},
+				},
+			},
+		},
+		RNGs: []libvirtxml.DomainRNG{
+			{
+				Model: "virtio",
+				Backend: &libvirtxml.DomainRNGBackend{
+					Random: &libvirtxml.DomainRNGBackendRandom{},
+				},
+			},
+		},
+		Consoles: []libvirtxml.DomainConsole{
+			{
+				Source: &libvirtxml.DomainChardevSource{
+					Pty: &libvirtxml.DomainChardevSourcePty{},
+				},
+				Target: &libvirtxml.DomainConsoleTarget{
+					Type: "virtio",
+					Port: &serialPort,
+				},
+			},
+		},
+	}
+
+	if runtime.GOARCH != "s390x" && runtime.GOARCH != "s390" {
```

Comment:
A short comment here for the s390 check would be great.

Reply:
> A short comment here for the s390 check would be great.

Done.

Reply:
I mean the comment saying why Graphics card is not being added for S390. :)

Reply:
Ah:), got it & a new comment added.